src/ContractInteract/EIP20Gateway.js

```diff
@@ -68,6 +97,244 @@ class EIP20Gateway {
     this.getValueTokenContract = this.getValueTokenContract.bind(this);
     this.isBountyAmountApproved = this.isBountyAmountApproved.bind(this);
     this.isStakeAmountApproved = this.isStakeAmountApproved.bind(this);
+    this.getStakeVault = this.getStakeVault.bind(this);
+    this.activateGateway = this.activateGateway.bind(this);
+    this.activateGatewayRawTx = this.activateGatewayRawTx.bind(this);
+  }
+
+  /**
+   * Setup for EIP20Gateway. Deploys EIP20Gateway and EIP20CoGateway,
+   * and links them.
+   *
+   * @param {Object} originWeb3 Web3 object.
+   * @param {Object} auxiliaryWeb3 Web3 object.
+   * @param {EIP20GatewaySetupConfig} gatewayConfig EIP20Gateway setup configuration.
+   * @param {EIP20CoGatewaySetupConfig} coGatewayConfig EIP20CoGateway setup configuration.
+   * @param {Object} originTxOptions Transaction options on Origin.
+   * @param {Object} auxiliaryTxOptions Transaction options on Auxiliary.
+   *
+   * @returns {Promise<EIP20Gateway>} Promise containing the EIP20Gateway instance that
+   *                                  has been set up.
+   */
+  static setup(
+    originWeb3,
+    auxiliaryWeb3,
```

Comment:
why not a `mosaic` object?

Reply:
I did attempt to change it to a `Mosaic` object, but there were a few things that didn't make that easy:

- Anchor is not directional, so at least there we would need to keep it as-is (though Anchor setup is supposed to move out `mosaic.js` anyway)
- When passing in a `Mosaic` object, does that only use it's `web3` objects, or does it also assume that the `contractAddresses` were set for the `Mosaic` object? Which contract addresses does it take from `Mosaic` and which should still be supplied via the `config`? How do we handle `baseToken` and `token` (as `contractAddresses` is supposed to have contract names as keys and they might potentially confilt here)? In the end all of that felt way too confusing and I decide it to leave it as separate `web3` objects.

I think `Mosaic` should only be for post-setup.